app/code/community/Bolt/Boltpay/Helper/Coupon.php

```diff
@@ -0,0 +1,684 @@
+
+        return "";
+    }
+
+    /**
+     * Gets the session quote if it exists.
+     *
+     * @return Mage_Sales_Model_Quote
+     * @throws Varien_Exception
+     */
+    protected function getParentQuote()
+    {
+        if (!$this->parentQuote) {
+            if (!$this->requestObject) {
+                return null;
+            }
+
+            /** @var Mage_Sales_Model_Quote $parentQuote */
+            $parentQuote = Mage::getModel('sales/quote')->loadByIdWithoutStore($this->getParentQuoteId());
+            $this->parentQuote = $parentQuote;
+        }
+
+        return $this->parentQuote;
+    }
+
+    /**
+     * Gets the immutable quote if it exists.
+     *
+     * @return Mage_Sales_Model_Quote
+     * @throws Varien_Exception
+     */
+    protected function getImmutableQuote()
+    {
+        if (!$this->immutableQuote) {
+            if (!$this->requestObject) {
+                return null;
+            }
+
+            /** @var Mage_Sales_Model_Quote $immutableQuote */
+            $immutableQuote = Mage::getModel('sales/quote')->loadByIdWithoutStore($this->getImmutableQuoteId());
+            $this->immutableQuote = $immutableQuote;
+        }
+
+        return $this->immutableQuote;
+    }
+
+    /**
+     * Get the coupon based on the coupon code if it exists.
+     *
+     * @return object
+     */
+    protected function getCoupon()
+    {
+        if (!$this->coupon) {
+            if (!$this->requestObject) {
+                return null;
+            }
+
+            $coupon = Mage::getModel('salesrule/coupon')->load($this->getCouponCode(), 'code');
+            $this->coupon = $coupon;
+        }
+
+        return $this->coupon;
+    }
+
+    /**
+     * Gets rule based on the coupon.
+     *
+     * @return object
+     */
+    protected function getRule()
+    {
+        if (!$this->rule) {
+            if (!$this->requestObject) {
+                return null;
+            }
+
+            $coupon = $this->getCoupon();
+            // Load the coupon discount rule
+            $rule = Mage::getModel('salesrule/rule')->load($coupon->getRuleId());
+            $this->rule = $rule;
+        }
+
+        return $this->rule;
+    }
+
+    /**
+     * Gets the coupon code from the request.
+     *
+     * @return string
+     */
+    protected function getCouponCode()
+    {
+        return @$this->requestObject->discount_code;
+    }
+
+    /**
+     * Gets increment id from the request.
+     *
+     * @return string
+     */
+    protected function getIncrementId()
+    {
+        // The latter two are transmitted as display_id field, separated by " | "
+        list($incrementId, $immutableQuoteId) = array_pad(
+            explode('|', @$this->requestObject->cart->display_id),
+            2,
+            null
+        );
+
+        return $incrementId;
+    }
+
+    /**
+     * Gets immutable quote id from the request.
+     *
+     * @return int
+     */
+    protected function getImmutableQuoteId()
```

Comment:
After further consideration, you should use: `Bolt_Boltpay_Helper_Api::getImmutableQuoteIdFromTransaction` and not introduce a new method unless you feel the need to wrap the call to this to simply pass it the parameter `$this->requestObject`.